test/mainnet/graph_reabse_fix.test.ts

```diff
@@ -0,0 +1,111 @@
+
+chai.use(solidity)
+const { expect } = chai
+
+describe('Graph Mainnet Fork Test - Rebase Fix', () => {
+  let GraphToken: ERC20
+  let Tenderizer: Graph
+  let tenderizerOwner: Signer
+
+  let signers: SignerWithAddress[]
+  let deployer: string
+  let cpBefore: BigNumber
+
+  before('get signers', async () => {
+    const namedAccs = await hre.getNamedAccounts()
+    signers = await ethers.getSigners()
+    deployer = namedAccs.deployer
+  })
+
+  const tenderizerAddr = '0xe66F3ab2f5621FE12ebf37754E1Af6d05b329A07'
+  const grtTokenAddress = '0xc944e90c64b2c07662a292be6244bdf05cda44a7'
+
+  const GRTHolder = '0xa64bc086d8bfaff4e05e277f971706d67559b1d1'
+  let GRTHolderSinger: Signer
+
+  const testTimeout = 120000
+
+  before('deploy Graph Tenderizer', async function () {
+    this.timeout(testTimeout)
+
+    // Fork from mainnet
+    await hre.network.provider.request({
+      method: 'hardhat_reset',
+      params: [
+        {
+          forking: {
+            block: 16448263,
+            jsonRpcUrl: process.env.ALCHEMY_MAINNET
+          }
+        }
+      ]
+    })
+
+    Tenderizer = (await ethers.getContractAt('Graph', tenderizerAddr)) as Graph
+
+    await hre.network.provider.request({
+      method: 'hardhat_impersonateAccount',
+      params: [GRTHolder]
+    })
+    GRTHolderSinger = await ethers.provider.getSigner(GRTHolder)
+
+    // Transfer some ETH
+    await hre.network.provider.send('hardhat_setBalance', [
+      GRTHolder,
+      `0x${ethers.utils.parseEther('100').toString()}`
+    ])
+
+    // Transfer some GRT
+    GraphToken = (await ethers.getContractAt('ERC20', grtTokenAddress)) as ERC20
+    await GraphToken.connect(GRTHolderSinger).transfer(deployer, ethers.utils.parseEther('100'))
+  })
+
+  describe('Rebase before upgrade', async () => {
+    it('reverts', async () => {
+      await expect(Tenderizer.claimRewards()).to.be.reverted
+    })
+  })
+
+  describe('Perform Upgrade', async function () {
+    before(async function () {
+      cpBefore = await Tenderizer.currentPrincipal()
+      const newFac = await ethers.getContractFactory('Graph', signers[0])
+      const newTenderizer = await newFac.deploy()
+      const proxy = (await ethers.getContractAt('EIP173Proxy', Tenderizer.address)) as EIP173Proxy
+      await hre.network.provider.send('hardhat_setBalance', [
+        await proxy.owner(),
+        `0x${ethers.utils.parseEther('100').toString()}`
+      ])
+
+      await hre.network.provider.request({
+        method: 'hardhat_impersonateAccount',
+        params: [await proxy.owner()]
+      })
+      tenderizerOwner = await ethers.provider.getSigner(await proxy.owner())
+      await proxy.connect(tenderizerOwner).upgradeTo(newTenderizer.address)
+    })
+
+    it('current priciple stays the same', async function () {
+      expect(await Tenderizer.currentPrincipal()).to.eq(cpBefore)
+    })
+  })
+
+  describe('Claim rewards succeeds', async function () {
```

Comment:
We should be able to calculate the exact reward amount and thus increase of currentprinciple

this needs more elaborate assertions